.github/workflows/full_build.yml

```diff
@@ -36,118 +35,244 @@ jobs:
+    - name: Store shared libs (/lib)
+      uses: actions/upload-artifact@v4
+      with:
+        name: lib
+        if-no-files-found: error
+        path: ${{runner.workspace}}/acados/lib/
+        compression-level: 0 #TODO test time difference
+        overwrite: true
+
+    - name: Store build scripts (/build)
+      uses: actions/upload-artifact@v4
+      with:
+        name: build
+        if-no-files-found: error
+        path: |
+          ${{runner.workspace}}/acados/build/
+          !${{runner.workspace}}/**/*.dir
+        compression-level: 0
+        overwrite: true
+        # exclude object files in .dir directories
+
+    - name: Store include directory (/include)
+      uses: actions/upload-artifact@v4
+      with:
+        name: include
+        if-no-files-found: error
+        path: ${{runner.workspace}}/acados/include/
+        compression-level: 0
+        overwrite: true
+
+  python_interface:
+    needs: core_build
+    runs-on: ubuntu-20.04
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
+
+    - uses: actions/download-artifact@v4
+      with:
+        path: ${{runner.workspace}}/acados
+
+    - name: Install Python interface
+      # TODO cache
       working-directory: ${{runner.workspace}}/acados
       shell: bash
       run: |
-        echo "ACADOS_SOURCE_DIR=$(pwd)" >> $GITHUB_ENV
-        echo "ACADOS_INSTALL_DIR=$(pwd)" >> $GITHUB_ENV
-        echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$(pwd)/lib" >> $GITHUB_ENV
-        echo "OCTAVE_PATH=$OCTAVE_PATH:$(pwd)/interfaces/acados_matlab_octave:$(pwd)/interfaces/acados_matlab_octave/acados_template_mex" >> $GITHUB_ENV
-        echo "MATLABPATH=$MATLABPATH:$(pwd)/interfaces/acados_matlab_octave:$(pwd)/interfaces/acados_matlab_octave/acados_template_mex" >> $GITHUB_ENV
+        echo "ACADOS_SOURCE_DIR=${{runner.workspace}}/acados" >> $GITHUB_ENV
+        echo "ACADOS_INSTALL_DIR=${{runner.workspace}}/acados" >> $GITHUB_ENV
+        echo "LD_LIBRARY_PATH=${{runner.workspace}}/acados/lib" >> $GITHUB_ENV
+        ${{runner.workspace}}/acados/.github/linux/install_python.sh''
 
-    # test interfaces
-    - name: Install Casadi MATLAB
-      working-directory: ${{runner.workspace}}/acados/external
+    - name: Run CMake python tests (ctest)
+
+      working-directory: ${{runner.workspace}}/acados/build
       shell: bash
-      run: |
-        sudo chmod +x ./../.github/linux/install_casadi_matlab.sh
-        ./../.github/linux/install_casadi_matlab.sh
-        echo "MATLABPATH=$MATLABPATH:$(pwd)/casadi-matlab" >> $GITHUB_ENV
+      # Execute tests defined by the CMake configuration.
+      # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
+      run: ${{runner.workspace}}/acados/.github/linux/run_tests.sh
 
-    - name: Prepare Octave
+  MATLAB_interface:
+    needs: core_build
+    runs-on: ubuntu-20.04
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
```

Comment:
I think this should not be needed for Simulink, Octave and Matlab.
For the Python test, there is a submodule with `test_data` which is some results are compared against.

Reply:
Checking out specific submodules seems to be not yet supported by the _checkout@v4_ action